src/app/pages/details/details.component.ts

```diff
@@ -14,25 +22,67 @@ import { takeUntilDestroyed } from '@angular/core/rxjs-interop';
   templateUrl: './details.component.html',
   styleUrl: './details.component.css'
 })
-export class DetailsComponent implements OnInit {
-  private route = inject(ActivatedRoute);;
-  private taskService = inject(TaskService);
+export class DetailsComponent implements OnInit{
+  private store: Store<AppState> = inject(Store);
   private destroyRef = inject(DestroyRef);
-  
-  public task$?: Observable<Task | undefined>;
+  private activatedRoute = inject(ActivatedRoute);
+  private router = inject(Router);
+
+  private taskId!: string;
+
+  public task$!: Observable<Task | undefined>;
+  public title!: string;
+  public desc!: string;
+  public deadline!: string;
 
   public ngOnInit(): void {
     this.initFields();
   }
 
+  public saveTask(event: Event): void {
+    event.preventDefault();
+
+    if (this.taskId === '') {
+      this.taskId = uuid();
+      this.deadline = this.mockDate();
+      this.store.dispatch((TaskActions['add']({task: this.buildTask()})));
+    } else {
+      this.store.dispatch((TaskActions['edit']({task: this.buildTask()})));
+    }
+    
+    this.router.navigate([Path.HOME]);
+  }
+
   private initFields(): void {
-    this.route.paramMap.pipe(
+    this.taskId = ''
```

Comment:
Please hold only one field of type Task instead of each property separately.
Then initialize this field here and remove `buildTask` method

Reply:
I used the taskId to determine whether this is a new task or an existing one being edited, an alternative would be to use a boolean for that and then initialize a Task field (with existing ID or generating new one), is that ok?

Reply:
i tried some different approach and I'd like to hear your opinion about it.

I still use a newTask method just for convenient purposes, but i now have a single Task object and i think its handled better on saveTask and on initFields

Reply:
Okay that's better, but please note that `subscribe` inside another `subscribe` is bad practice in Angular. 
Please find a way to use RxJS operators for preventing this.

Reply:
switchMap of course >,<
I'll add this update to the Optimizations PL